aptos-move/framework/aptos-framework/sources/stake.move

```diff
@@ -511,6 +518,26 @@ module aptos_framework::stake {
         let amount = coin::value<AptosCoin>(&coins);
         assert!(amount > 0, error::invalid_argument(EINVALID_STAKE_AMOUNT));
 
+        // Only track and validate voting power increase for active and pending_active validator.
+        // Pending_inactive validator will be removed from the validator set in the next epoch.
+        // Inactive validator's total stake will be tracked when they join the validator set.
+        let validator_state = get_validator_state(pool_address);
+        if (validator_state == VALIDATOR_STATUS_ACTIVE || validator_state == VALIDATOR_STATUS_PENDING_ACTIVE) {
+            let validator_set = borrow_global_mut<ValidatorSet>(@aptos_framework);
+
+            // Only validate if current total voting power is > 0.
+            let total_voting_power = validator_set.total_voting_power;
+            if (total_voting_power > 0) {
```

Comment:
Is this if-statement necessary? It seems that this condition should be always true because there should be at least one active validator with the minimum stake.

Reply:
No this was required for a test because there's only 1 validator (the validator set starts out empty). I removed this check since we shouldn't have one just for tests.